### PR TITLE
Feature/BC-1138 - Import course on room page

### PR DESCRIPTION
**Pull request by murat-merdoglu-hpi**
# Short Description
- Import a course feature is added on room-overview page

## Links to Ticket and related Pull-Requests
https://ticketsystem.hpi-schul-cloud.org/browse/BC-1138

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/molecules/ImportModal.vue [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	computed: {
+		nextButtonName() {
+			if (this.step < 3)
+				return this.$t("pages.rooms.importCourse.btn.continue");
+			return this.$t("pages.rooms.importCourse.btn.confirm");
+		},
+	},
+	methods: {
+		nextStep() {
+			if (this.step == 2) {
```
Comment (MartinSchuhmacher) [inline src/components/molecules/ImportModal.vue:147]: === instead?

File: src/components/molecules/ImportModal.vue [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+				if (this.sharedCourseData.code === "") return;
+				this.getSharingStatus(this.sharedCourseData.code);
+			}
+			if (this.step == 3) {
```
Comment (MartinSchuhmacher) [inline src/components/molecules/ImportModal.vue:151]: === instead?

File: src/components/molecules/ImportModal.vue [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+				this.$emit("dialog-closed", false);
+				window.location = `/courses/${importedCourseId}/edit/`;
+			}
+		},
+		cancel() {
+			this.$emit("dialog-closed", false);
+			this.step = 1;
+		},
+		stepBack() {
+			if (this.step == 1) {
```
Comment (MartinSchuhmacher) [inline src/components/molecules/ImportModal.vue:177]: === instead?

File: src/store/rooms.unit.ts [hunk 4]
```diff
@@ -9,8 +9,9 @@ let getRequestReturn: any = {};
 
 const axiosInitializer = () => {
 	initializeAxios({
-		$get: async (path: string) => {
+		$get: async (path: string, data?: object) => {
 			receivedRequests.push({ path });
+			receivedRequests.push({ data });
```
Comment (MartinSchuhmacher) [inline src/store/rooms.unit.ts:14]: can this be done in one push?

**Review by MartinSchuhmacher — CHANGES_REQUESTED** [review]

File: locale/en.json [hunk 5]
```diff
@@ -666,6 +666,20 @@
...
 	"pages.rooms.headerSection.ariaLabel": "List of current courses {itemCount} elements",
 	"pages.room.taskCard.aria": "{itemType}, link, {itemName}, press enter to open",
+	"pages.rooms.fab.add": "Create",
+	"pages.rooms.fab.add.aria": "Create course",
+	"pages.rooms.fab.import": "Import",
+	"pages.rooms.fab.import.aria": "Import course",
+	"pages.rooms.importCourse.step_1.text": "Info",
+	"pages.rooms.importCourse.step_2.text": "Paste the code",
+	"pages.rooms.importCourse.step_3.text": "Course name",
+	"pages.rooms.importCourse.step_1.info_1": "A course folder is automatically created for the imported course. Student-related data from the original course will be removed. Then add students and make an appointment for the course.",
```
[the remark this answers is not in the document]
Comment (murat-merdoglu-hpi) [reply locale/en.json:676]: It is as same as on the legacy client importing course.

**Review by MartinSchuhmacher — APPROVED** [review]

**Pull request merged**